### DISCUSSION TITLE
Color Field with fixed/static Color-Selection

### DISCUSSION THREAD

**Discussion by alexanderbuergin** [craftcms/cms#15107] [Ideas] [answered]
It would be beneficial to add an option in the color picker field that allows only predefined colors to be selected, instead of allowing any arbitrary hex value. This could be implemented as a dropdown list with the available colors. This feature would enhance consistency and user-friendliness by limiting the selection to a set color palette.

**Example:**
Instead of entering any arbitrary hex format, the user can only choose from the following colors:
- Red (#FF0000)
- Green (#00FF00)
- Blue (#0000FF)
- Yellow (#FFFF00)
- etc.

**Benefits:**
- Increased consistency in color selection
- Easier handling for the user
- Prevention of errors from entering invalid hex values

**Implementation:**
A possible implementation could include an additional option in the configuration menu where the predefined colors can be set. The color picker field would then display a dropdown menu from which the user can select the desired color.

**Current Options:**
<img width="222" alt="Bildschirmfoto 2024-05-30 um 15 26 01" src="https://github.com/craftcms/cms/assets/85960512/4d0ca3bd-2b89-43e9-99e7-ac69a1aec2db">

**Possible implementation:**
<img width="912" alt="Bildschirmfoto 2024-05-30 um 15 49 56" src="https://github.com/craftcms/cms/assets/85960512/435c56e2-9f19-40f6-bdaa-5a195d1b096a">

Add a light switch:

<img width="621" alt="Bildschirmfoto 2024-05-30 um 15 56 49" src="https://github.com/craftcms/cms/assets/85960512/c0e87f4b-85bc-480a-b006-8885e2f58a03">

**Comment by piotrpog** [comment]
There is color swatches plugin right now that handles that. But i think it should be build-in into craft of at least made first party plugin. I wrote about it some time ago:
https://github.com/craftcms/cms/discussions/14488

@brandonkelly 
Are there perhaps any plans for this?

**Comment by brandonkelly** [comment, marked as answer]
Got around to this for Craft 5.6! (#16249)

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out with this 🎉

**Comment by piotrpog** [comment]
@brandonkelly Thanks a lot! This is great news.
